CalibTracker/SiStripHitEfficiency/plugins/SiStripHitEfficiencyHarvester.cc

```diff
@@ -816,45 +809,86 @@ void SiStripHitEfficiencyHarvester::setEffBinLabels(const T gr, const T gr2, con
+  const auto& folderString = ::projFolder[theProj];
+  const auto& foundHistoString = ::projFoundHisto[theProj];
+  const auto& totalHistoString = ::projTotalHisto[theProj];
+  const auto& titleString = ::projTitle[theProj];
+  const auto& titleXString = ::projXtitle[theProj];
+
+  LogDebug("SiStripHitEfficiencyHarvester")
+      << " inside" << __PRETTY_FUNCTION__ << " from " << ::projFolder[theProj] << " " << __LINE__ << std::endl;
 
-void SiStripHitEfficiencyHarvester::makeSummaryVsLumi(DQMStore::IGetter& getter) const {
   for (unsigned int iLayer = 1; iLayer != (showRings_ ? 20 : 22); ++iLayer) {
-    auto hfound = getter.get(fmt::format("{}/VsLumi/layerfound_vsLumi_layer_{}", inputFolder_, iLayer))->getTH1F();
-    auto htotal = getter.get(fmt::format("{}/VsLumi/layertotal_vsLumi_layer_{}", inputFolder_, iLayer))->getTH1F();
+    LogDebug("SiStripHitEfficiencyHarvester")
+        << "iLayer " << iLayer << " " << fmt::format("{}/{}/{}{}", inputFolder_, folderString, foundHistoString, iLayer)
+        << std::endl;
+
+    const auto lyrName = ::layerName(iLayer, showRings_, nTEClayers_);
+    auto hfound = getter.get(fmt::format("{}/{}/{}{}", inputFolder_, folderString, foundHistoString, iLayer));
+    auto htotal = getter.get(fmt::format("{}/{}/{}{}", inputFolder_, folderString, totalHistoString, iLayer));
 
     if (hfound == nullptr or htotal == nullptr) {
       if (hfound == nullptr)
         edm::LogError("SiStripHitEfficiencyHarvester")
-            << fmt::format("{}/VsLumi/layerfound_vsLumi_layer_{}", inputFolder_, iLayer) << " was not found!";
+            << fmt::format("{}/{}/{}{}", inputFolder_, folderString, foundHistoString, iLayer) << " was not found!";
       if (htotal == nullptr)
         edm::LogError("SiStripHitEfficiencyHarvester")
-            << fmt::format("{}/VsLumi/layertotal_vsLumi_layer_{}", inputFolder_, iLayer) << " was not found!";
+            << fmt::format("{}/{}/{}{}", inputFolder_, folderString, totalHistoString, iLayer) << " was not found!";
       // no input histograms -> continue in the loop
       continue;
     }
 
-    if (!hfound->GetSumw2())
-      hfound->Sumw2();
-    if (!htotal->GetSumw2())
-      htotal->Sumw2();
-    for (Long_t i = 0; i != hfound->GetNbinsX() + 1; ++i) {
-      if (hfound->GetBinContent(i) == 0)
-        hfound->SetBinContent(i, 1e-6);
-      if (htotal->GetBinContent(i) == 0)
-        htotal->SetBinContent(i, 1);
+    // in order to display correct errors when taking the ratio
+    if (!hfound->getTH1F()->GetSumw2())
+      hfound->getTH1F()->Sumw2();
+    if (!htotal->getTH1F()->GetSumw2())
+      htotal->getTH1F()->Sumw2();
+
+    // prevent dividing by 0
+    for (int i = 0; i != hfound->getNbinsX() + 1; ++i) {
+      if (hfound->getBinContent(i) == 0)
+        hfound->setBinContent(i, 1e-6);
+      if (htotal->getBinContent(i) == 0)
+        htotal->setBinContent(i, 1);
```

Comment:
I'm not sure I understand this. If the bin content is 0, set it to 1? Why not leave it at zero?

Reply:
It might lead to divisions by zero.